domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1596,91 +1595,217 @@ public TotaledResults(ResultSet set, Integer total) {
 
-    client.getConnection(res -> {
-      if (res.succeeded()) {
-        SQLConnection connection = res.result();
-        Class<T> clazz = (Class<T>) entity.getClass();
-        doStreamGet(connection, false, table, clazz, fieldName, filter, returnIdField, distinctOn,
-            Collections.emptyList(), streamHandler, replyHandler);
-      } else {
-        replyHandler.handle(Future.failedFuture(res.cause()));
-      }
-    });
+    streamGet(table, clazz, fieldName, filter, returnIdField, distinctOn,
+      Collections.emptyList(), replyHandler);
   }
 
-  @SuppressWarnings({"unchecked", "squid:S00107"})
-  private <T> void doStreamGet(SQLConnection connection, boolean transactionMode, String table, Class<T> clazz,
-      String fieldName, CQLWrapper wrapper, boolean returnIdField, String distinctOn, List<FacetField> facets,
-      Handler<T> streamHandler, Handler<AsyncResult<Void>> replyHandler) {
+  /**
+   * Stream GET with CQLWrapper and facets {@link org.folio.rest.persist.PostgresClientStreamResult}
+   * @param <T>
+   * @param table
+   * @param clazz
+   * @param fieldName
+   * @param filter
+   * @param returnIdField must be true if facets are in passed
+   * @param distinctOn may be null
+   * @param facets may not be null (Collections.emptyList() for no facets)
+   * @param replyHandler AsyncResult; on success with result {@link org.folio.rest.persist.PostgresClientStreamResult}
+   */
+  @SuppressWarnings({"unchecked", "squid:S00107"})    // Method has >7 parameters
+  public <T> void streamGet(String table, Class<T> clazz, String fieldName,
+    CQLWrapper filter, boolean returnIdField, String distinctOn,
+    List<FacetField> facets, Handler<AsyncResult<PostgresClientStreamResult<T>>> replyHandler) {
 
-    vertx.runOnContext(v1 -> {
-      try {
-        QueryHelper queryHelper = buildQueryHelper(transactionMode, table, fieldName, wrapper, returnIdField, facets,
-            distinctOn);
+    client.getConnection(conn ->
+      doStreamGet(conn, table, clazz, fieldName, filter, returnIdField,
+        distinctOn, facets, closeAndHandleResult(conn, replyHandler)));
+  }
 
-        connection.queryStream(queryHelper.selectQuery, stream -> {
-          if (stream.succeeded()) {
+  /**
+   * private for now, might be public later (and renamed)
+   * @param <T>
+   * @param connResult
+   * @param table
+   * @param clazz
+   * @param fieldName
+   * @param wrapper
+   * @param returnIdField
+   * @param distinctOn
+   * @param facets
+   * @param replyHandler
+   */
+  @SuppressWarnings({"unchecked", "squid:S00107"})    // Method has >7 parameters
+  private <T> void doStreamGet(AsyncResult<SQLConnection> connResult,
+    String table, Class<T> clazz, String fieldName, CQLWrapper wrapper,
+    boolean returnIdField, String distinctOn, List<FacetField> facets,
+    Handler<AsyncResult<PostgresClientStreamResult<T>>> replyHandler) {
+    if (connResult.failed()) {
```

Comment:
Can you insert a blank line to separate the argument list from the code?

Reply:
Good catch. I think, in general, it's a good idea - for multiline declarations like that to have a blank line.